DEMO PAGE

akanellop opened the pull request.

akanellop committed:
Index.html runs a demo page for our program ,demonstrating how its output could look in html format using a browser, it actually opens a page in which you can enter your text to test .env.spec program.
demoPage.js contains the javascript script needed for the index.html , actually contains a simple function for separating the two outputs , one in html code and the actual appearance of the html
styles.css contains some css formatting for the index.html file

akanellop commented:
![screenshot from 2018-07-26 15-32-05](https://user-images.githubusercontent.com/24398390/43262548-3120b724-90e9-11e8-9fca-57a1c4db1549.png)

Screenshot of demo page

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <title>Document</title>
```
Rename this to `.env.spec demo` please.

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+<!DOCTYPE html>
```
Move this file to `demo/index.html` please.

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+
+  <body>
+    <script type="text/javascript" src="dist/demoPage.js"></script>
+    <p><h1>.ENV.SPEC demo page</h1></p>
```
1. Remove the parent `<p>` tag, as `<h1>` tags cannot be put inside paragraphs.
1. Rename `.ENV.SPEC` to `.env.spec`

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+  <body>
+    <script type="text/javascript" src="dist/demoPage.js"></script>
+    <p><h1>.ENV.SPEC demo page</h1></p>
+    <p><h4><i>Type or paste your .env.spec file in the text area below :</i></h4></p>
```
1. Remove the parent `<p>` tag, as `<h4>` tags cannot be put inside paragraphs.
1. Remove the `<i>` tag. We will apply styling later with CSS

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+  </head>
+
+  <body>
+    <script type="text/javascript" src="dist/demoPage.js"></script>
```
1. This should be moved down, in the line above the closing tag of body (above `</body>`).
2. This file is now ignored, since it is inside the `dist` directory, move it to `demo/main.js` to track it in Git

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+        <textarea id="outputArea" placeholder="HTML text" rows = "8" cols = "60" readonly></textarea>
+    </div>
+
+    <p><h4><i>HTML RESULT :</i></h4></p>
```
1. Remove the parent `<p>` tag, as `<h4>` tags cannot be put inside paragraphs.
2. Remove the `<i>` tag. We will apply styling later with CSS
3. Rename `HTML RESULT :` to `Rendered form:`

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+    <p><h4><i>HTML RESULT :</i></h4></p>
+    <div id="finalOutput"></div>
+
+    <script>
```
Remove this empty `script` tag. It does nothing here.

parisk commented on index.html:
```diff
@@ -0,0 +1,29 @@
+        <textarea id="inputArea" placeholder="Start here..." rows = "6" cols = "30"></textarea>
+        <button value="Get HTML" onclick="getHTML()"> Get HTML</button>
+    </div>
+    <div>
```
Can you add a heading (I guess `h4`) above here stating that this is `HTML output: `?

parisk reviewed (changes requested):
Great! First batch of updates are here.

I also suggest squashing the current commits to one.

akanellop committed:
Demo file contains all demo documents such as js, html and css .
Index.html doesnt work, error :cannot GET/ .
index.html and packageJson may need changes.

akanellop committed:
Index.html working
demo page need som fixes with css
changed package.json so we can use both demo and dist directories(dist not showing in git)

akanellop commented:
![demo](https://user-images.githubusercontent.com/24398390/43396282-c5ab6cd6-9409-11e8-8501-df6dd0ffbbc5.png)

parisk commented on demo/index.html:
```diff
@@ -0,0 +1,26 @@
+
+    <div>
+        <textarea id="inputArea" placeholder="Start here..." rows = "6" cols = "30"></textarea>
+        <button value="Get HTML" onclick="getHTML()"> Get HTML</button>
```
Instead of using the `onclick` HTML attribute, we should do this with `addEventListener` in plain JavaScript.

parisk commented on demo/index.html:
```diff
@@ -0,0 +1,26 @@
+        <textarea id="outputArea" placeholder="HTML text" rows = "8" cols = "60" readonly></textarea>
+    </div>
+
+    <h4>HTML RESULT :</h4>
```
I think we should prefer a more descriptive heading here. What about **Rendered result**?

parisk commented on demo/index.html:
```diff
@@ -0,0 +1,26 @@
+        <textarea id="inputArea" placeholder="Start here..." rows = "6" cols = "30"></textarea>
+        <button value="Get HTML" onclick="getHTML()"> Get HTML</button>
+    </div>
+    <div>
```
Add a heading (I guess `h4`) here as well please. Something simple like **HTML output** can do the work.

parisk commented on demo/main.css:
```diff
@@ -0,0 +1,38 @@
+body {
+    background-color: #E4E4E5;
```
Can you remove the `background-color` please? We prefer leaving simple pages with white background as it is less tiring for the eye when reading.

Also, can you add the `sans-serif` font family?

parisk commented on demo/main.css:
```diff
@@ -0,0 +1,38 @@
+    background-color: #E4E4E5;
+}
+h1 {
+    color: black;
```
Can you remove this please, as we do not need it? The color is black by default 😁.

parisk commented on demo/main.css:
```diff
@@ -0,0 +1,38 @@
+    margin-bottom: 20px;
+}
+
+input{
```
What I would like to see also is displaying labels, inputs (including text areas) and buttons as `block` elements.

akanellop committed:
Changes
index.html : each element in block tag such as h4,h1,pre,div,
 removed onclick function for button,changed variables' names
main.css : removed unnecessary blocks,changed style of page
main.js : improvement with addEventListener function for button

index.html still a working demo page for .env.spec program

akanellop committed:
in main.css insertion of block input so the rendered results are alligned

parisk commented on package.json:
```diff
@@ -9,11 +9,17 @@
+    "start": "browser-sync start --server demo --server dist --watch --files 'dist/*' 'demo/*'  --port $PORT"
+  },
+  "dependencies": {
+    "browser-sync": "^2.24.6"
```
We need `browser-sync` to be in the `devDependencies`, since it is a tool used by the developers of `env-spec-js` (us) and is not necessary for the users of `env-spec-js` itself.

You can fix this by running the following commands:

```
yarn remove browser-sync
yarn add --dev browser-sync
```

akanellop committed:
changed browser-sync to dev browser sync

parisk reviewed (approved):
That's great, thanks!
